js-exercises/duplicate-letters/duplicateLetters.js

```diff
@@ -0,0 +1,49 @@
+const isEmptyString = (argument) => {
+  if (typeof argument !== 'string') {
+    throw TypeError(`expects string input, and input has ${typeof argument}`);
+  }
+
+  if (argument.trim().length < 1) {
+    return true;
+  }
+
+  return false;
+};
+
+const letterRegex = /^[A-Za-z]+$/;
+
+const isLetter = (letter) => letterRegex.test(letter);
+
+function duplicateLetters(argument) {
+  if (isEmptyString(argument)) {
+    return false;
+  }
+
+  const letters = [...argument];
+  const uniqueLetters = [...new Set(letters)];
+
+  const mostDuplicateLetter = { count: 0, letter: null };
+
+  for (const letter of uniqueLetters) {
+    if (!isLetter(letter)) {
+      throw Error(`Expects english characters only, and input has ${letter}`);
+    }
+
+    const letterCount = letters.filter(
+      (filteredLetter) => filteredLetter === letter
+    ).length;
+
+    if (letterCount > mostDuplicateLetter.count) {
+      mostDuplicateLetter.count = letterCount;
+      mostDuplicateLetter.letter = letter;
```

Comment:
Do we really need to store the letter as well? Just trying to understand your thought process. Were you trying to implement thinking that in the future you might need it?

Reply:
This program just needed count, but I went with real time expectation to keep letter as well in program.
Will remove from here for the time, and keep it fully according to specs

Reply:
it's removed with https://github.com/pesto-students/deep-dive-varora1406/pull/1/commits/f47a8012116b60b63ef809392c344e14ec1e118f